pkg/clustermanager/cluster_manager.go

```diff
@@ -890,8 +892,10 @@ func (c *ClusterManager) PauseEKSAControllerReconcile(ctx context.Context, clust
 		if clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef == nil {
 			return fmt.Errorf("machineGroupRef for control plane is not defined")
 		}
-		if len(clusterSpec.Spec.WorkerNodeGroupConfigurations) <= 0 || clusterSpec.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef == nil {
-			return fmt.Errorf("machineGroupRef for worker nodes is not defined")
+		for _, workerNodeGroupConfiguration := range clusterSpec.Spec.WorkerNodeGroupConfigurations {
+			if workerNodeGroupConfiguration.MachineGroupRef == nil {
```

Comment:
nit: this almost seems like a validation, if we don't have it at the api validations level, we could add it. And remove it from here